cmd/buildah/rmi.go

```diff
@@ -47,26 +57,58 @@ func rmiCmd(c *cli.Context) error {
 		return err
 	}
 
-	for _, id := range args {
-		image, err := getImage(id, store)
+	imagesToDelete := args[:]
+	var lastError error
+
+	if removeAll {
+		images, err := store.Images()
 		if err != nil {
-			return errors.Wrapf(err, "could not get image %q", id)
+			return errors.Wrapf(err, "error reading images")
+		}
+		for _, image := range images {
+			imagesToDelete = append(imagesToDelete, image.ID)
+		}
+	}
+
+	for _, id := range imagesToDelete {
+		image, err := getImage(id, store)
+		if err != nil || image == nil {
+			if lastError != nil {
+				fmt.Fprintln(os.Stderr, lastError)
+			}
+			if err == nil {
+				err = errors.New("Image does not Exist")
+			}
+			lastError = errors.Wrapf(err, "could not get image %q", id)
+			continue
 		}
 		if image != nil {
 			ctrIDs, err := runningContainers(image, store)
 			if err != nil {
-				return errors.Wrapf(err, "error getting running containers for image %q", id)
+				if lastError != nil {
+					fmt.Fprintln(os.Stderr, lastError)
+				}
+				lastError = errors.Wrapf(err, "error getting running containers for image %q", id)
+				continue
 			}
 			if len(ctrIDs) > 0 && len(image.Names) <= 1 {
 				if force {
 					err = removeContainers(ctrIDs, store)
 					if err != nil {
-						return errors.Wrapf(err, "error removing containers %v for image %q", ctrIDs, id)
+						if lastError != nil {
```

Comment:
Should we check if the error was container is not running?  Or does removeContainers take care of this for us?  Otherwise we have a race condition.

Reply:
removeContainers is a stub that calls store.DeleteContainers.  That appears to delete containers without a "running" check.  I'll let @nalind slap me around if I'm off-base.

I setup a registry that was running in a container and did "buildah rmi --all --force" and it wiped out the images and the associated container including the one running the registry.  However the registry was still up but not functioning very happily until I <ctrl>-c'd it.

I setup  another container that printed to stdout, slept a second, then looped back to the print.  After I removed the container, the prints kept happening but the sleeps did not as /usr/sbin/sleep returned "No such file or directory".  It died by itself when the loop finished up

So long story short, we delete running containers, they're left in a haphazard state until they die.  Do we need to adjust?  Personally I think perhaps a stronger warning in the man page would suffice.

If you're interested, here's the rmi process for the registry below.

```
# buildah images
IMAGE ID             IMAGE NAME                                               CREATED AT             SIZE
177391bcf802         docker.io/library/registry:2                             Dec 1, 2017 22:15      34.1 MB
[root@docker buildah]# buildah containers
CONTAINER ID  BUILDER  IMAGE ID     IMAGE NAME                       CONTAINER NAME
9fad5cae923a     *                  scratch                          working-container
5638b62b89ec     *     177391bcf802 docker.io/library/registry:2     registry-working-container
[root@docker buildah]# buildah rmi -f --all
177391bcf8024d4db90069c89920576acd00a4eaba72e53ca0dc8e9a7501f093
[root@docker buildah]# buildah images
[root@docker buildah]# buildah containers
CONTAINER ID  BUILDER  IMAGE ID     IMAGE NAME                       CONTAINER NAME
9fad5cae923a     *                  scratch                          working-container
```